plenum/server/request_handlers/node_handler.py

```diff
@@ -1,17 +1,200 @@
+from plenum.common.types import f
+from plenum.server.database_manager import DatabaseManager
 from plenum.server.request_handlers.handler_interfaces.write_request_handler import WriteRequestHandler
+from plenum.server.request_handlers.utils import is_steward
 
 
 class NodeHandler(WriteRequestHandler):
+    state_serializer = pool_state_serializer
+
+    def __init__(self, database_manager: DatabaseManager, bls_crypto_verifier):
+        super().__init__(database_manager, NODE, POOL_LEDGER_ID)
+        self.bls_crypto_verifier = bls_crypto_verifier
 
     def static_validation(self, request: Request):
-        pass
+        self._validate_request_type(request)
+        blskey = request.operation.get(DATA).get(BLS_KEY, None)
+        blskey_proof = request.operation.get(DATA).get(BLS_KEY_PROOF, None)
+        if blskey is None and blskey_proof is None:
+            return
+        if blskey is None and blskey_proof is not None:
+            raise InvalidClientRequest(request.identifier, request.reqId,
+                                       "A Proof of possession is not "
+                                       "needed without BLS key")
+        if blskey is not None and blskey_proof is None:
+            raise InvalidClientRequest(request.identifier, request.reqId,
+                                       "A Proof of possession must be "
+                                       "provided with BLS key")
+        if not self._verify_bls_key_proof_of_possession(blskey_proof,
+                                                        blskey):
+            raise InvalidClientRequest(request.identifier, request.reqId,
+                                       "Proof of possession {} is incorrect "
+                                       "for BLS key {}".
+                                       format(blskey_proof, blskey))
+
+    def dynamic_validation(self, request: Request):
+        self._validate_request_type(request)
+        node_nym = request.operation.get(TARGET_NYM)
+        if self._get_node_data(node_nym, is_committed=False):
+            error = self._auth_error_while_updating_node(request)
+        else:
+            error = self._auth_error_while_adding_node(request)
+        if error:
+            raise UnauthorizedClientRequest(request.identifier, request.reqId,
+                                            error)
+
+    def _update_state_with_single_txn(self, txn, is_committed=True):
+        self._validate_txn_type(txn)
+        node_nym = get_payload_data(txn).get(TARGET_NYM)
+        data = get_payload_data(txn).get(DATA, {})
+        existing_data = self._get_node_data(node_nym, is_committed=is_committed)
+        # Node data did not exist in state, so this is a new node txn,
+        # hence store the author of the txn (steward of node)
+        if not existing_data:
+            existing_data[f.IDENTIFIER.nm] = get_from(txn)
+        existing_data.update(data)
+        key = node_nym.encode()
```

Comment:
Do we have `gen_txn_path` implemented here?

Reply:
fixed